velox/functions/prestosql/aggregates/tests/AverageAggregationTest.cpp

```diff
@@ -534,5 +534,80 @@ TEST_F(AverageAggregationTest, constantVectorOverflow) {
   assertQuery(plan, "SELECT 1073741824");
 }
 
+TEST_F(AverageAggregationTest, companionFunctionsWithNonFlatInputs) {
+  auto testFunction = [this](const std::string& functionName) {
+    auto indices = makeIndices({0, 1, 2, 3, 4});
+    VectorPtr row = makeRowVector(
+        {BaseVector::wrapInDictionary(
+             nullptr,
+             indices,
+             5,
+             makeFlatVector<double>({1.0, 2.0, 3.0, 4.0, 5.0})),
+         makeFlatVector<int64_t>({1, 1, 1, 1, 1})});
+    auto rowInDict = BaseVector::wrapInDictionary(nullptr, indices, 5, row);
+
+    auto sumVector = std::make_shared<LazyVector>(
+        pool(),
+        DOUBLE(),
+        5,
+        std::make_unique<velox::test::SimpleVectorLoader>([&](auto rows) {
+          return makeFlatVector<double>({1.0, 2.0, 3.0, 4.0, 5.0});
+        }));
+    VectorPtr row2 = makeRowVector({
+        sumVector,
+        BaseVector::wrapInDictionary(
+            nullptr,
+            indices,
+            5,
+            BaseVector::wrapInConstant(5, 0, makeFlatVector<int64_t>({1, 2}))),
+    });
+    auto key = makeFlatVector<bool>({true, false, true, false, true});
+    auto input = makeRowVector({key, rowInDict, row2});
```

Comment:
can you add a small comment about the composition of rowInDict and row2 something like Dictionary(ROW(Flat, Flat)) and ROW(Lazy, Dictionary(Constant(flat)). Maybe also mention in the test name that we are testing lazy too, not just non-flat